packages/toolkit/core-kit/src/password-policy.ts

```diff
@@ -282,171 +313,143 @@ export class PasswordPolicyChecker {
   }
 
   /**
-   * Check if the given password contains repetition characters that are more than
-   * the threshold.
+   * Get the length of the repetition at the beginning of the given string.
+   * For example, `repetitionLength('aaaaa')` will return `5`.
    *
-   * @param password - Password to check.
-   * @returns Whether the password contains repetition characters.
+   * If the length is less than {@link PasswordPolicyChecker.repetitionAndSequenceThreshold},
+   * `0` will be returned.
    */
-  hasRepetition(password: string): boolean {
-    const { repetitionAndSequenceThreshold, getRestrictedPhraseThreshold } = PasswordPolicyChecker;
+  /* eslint-disable @silverhand/fp/no-let, @silverhand/fp/no-mutation */
+  repetitionLength(password: string): number {
+    const { repetitionAndSequenceThreshold } = PasswordPolicyChecker;
+    const firstChar = password[0]?.toLowerCase();
+    let length = 0;
 
-    if (password.length < repetitionAndSequenceThreshold) {
-      return false;
+    for (const char of password) {
+      if (char.toLowerCase() === firstChar) {
+        length += 1;
+      } else {
+        break;
+      }
     }
 
-    const matchedRepetition = password.match(
-      new RegExp(`(.)\\1{${getRestrictedPhraseThreshold(password)},}`, 'g')
-    );
-
-    return Boolean(matchedRepetition);
+    return length >= repetitionAndSequenceThreshold ? length : 0;
   }
+  /* eslint-enable @silverhand/fp/no-let, @silverhand/fp/no-mutation */
 
   /**
-   * Check if the given password contains user information.
+   * Get the length of the user information at the beginning of the given string.
+   * For example, `userInfoLength('silverhand', { username: 'silverhand' })` will return `10`.
    *
-   * @param password - Password to check.
-   * @param userInfo - User information to check.
-   * @returns Whether the password contains user information.
+   * For multiple matches, the longest length will be returned.
    */
   // eslint-disable-next-line complexity
-  hasUserInfo(password: string, userInfo: UserInfo): boolean {
-    const lowercasedPassword = password.toLowerCase();
+  userInfoLength(password: string, userInfo: UserInfo): number {
```

Comment:
we only check whether the user info is located at the beginning of the password?

Reply:
This is the function is used for. See `check()` for complete verification.